proto/dwtools/abase/l1.test/Str.test.s

```diff
@@ -8835,6 +8836,80 @@ ghij
+  expected = 'f';
+  test.identical( got.nearest[ 0 ], expected );
+
+  test.case = 'charsRange[0] a few letters';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 7 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'funct';
+  test.identical( got.nearest[ 0 ], expected );
+
+  test.case = 'charsRange[1] one letter';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 6 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'i';
+  test.identical( got.nearest[ 1 ], expected );
+
+  test.case = 'charsRange[1] a few letters';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 1 });
+  // console.log( got );
+  expected = 'ion';
+  test.identical( got.nearest[ 1 ], expected );
+
+  test.case = 'one line report';
+  got = _.strLinesNearestReport({ src : 'function add( x,y ) { return x + y }', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 1 });
+  expected = 'function add( x,y ) { return x + y }';
+  test.identical( got.report, '1 : ' + expected );
+
+  test.case = 'multiline report';
+  got = _.strLinesNearestReport({ src : 'a\nb\nc\ndef\ng\nk', charsRangeLeft : [ 3, 5 ], gray : 1, numberOfLines : 5 });
+  // console.log( got );
+  expected = '3 : a\n4 : b\n5 : c\n6 : def\n7 : g';
+  test.identical( got.report, expected );
+
+  test.case = 'number of lines: out of the range';
+  got = _.strLinesNearestReport({ src : 'test0\nfunction add( x,y ) { return x + y }\ntest2\ntest3\ntest4\n test5', charsRangeLeft : [ 5, 8 ], gray : 1, numberOfLines : 2 });
+  // console.log( got );
+  expected = 'nction add( x,y ) { return x + y }';
+  test.identical( got.nearest[ 2 ], expected );
+
+  test.case = 'number of lines: in the range';
+  got = _.strLinesNearestReport({ src : '0\nabcde\n1\n2\n3', charsRangeLeft : [ 3, 5 ], gray : 1, numberOfLines : 4 });
+  // console.log( got );
+  expected = 'de\n1';
+  test.identical( got.nearest[ 2 ], expected );
+
+  test.case = 'lines to the left of the range';
+  got = _.strLinesNearestReport({ src : '0\nabcde\n1\n2\n3', charsRangeLeft : [ 3, 5 ], gray : 1, numberOfLines : 4 });
+  // console.log( got );
+  expected = '0\na';
+  test.identical( got.nearest[ 0 ], expected );
+
+  test.case = 'lines to the right of the range';
+  got = _.strLinesNearestReport({ src : '0\nabcde\n1\n2\n3', charsRangeLeft : [ 3, 5 ], gray : 1, numberOfLines : 4 });
+  // console.log( got );
+  expected = 'de\n1';
+  test.identical( got.nearest[ 2 ], expected );
+
+  test.case = 'a lot of lines';
+  got = _.strLinesNearestReport({ src : 'a\nb\nc\ndef\ng\nk', charsRangeLeft : [ 6, 9 ], gray : 1, numberOfLines : 5 });
+  // console.log( got );
+  expected = 'def';
+  test.identical( got.nearest[ 1 ], expected );
+  test.identical( got.nearest[ 0 ], 'b\nc\n' );
+  test.identical( got.nearest[ 2 ], '\ng\nk' );
```

Comment:
If exists exceptions ( see `_.assert` in  routine ), add throwing test cases, please.
https://github.com/Wandalen/wTools/blob/master/proto/dwtools/abase/l1.test/Str.test.s#L883